From 18f78741766705e5035bc9325a0ce20b5f768f45 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Tue, 25 Sep 2012 19:11:28 +0900
Subject: [PATCH 67/97] ext/dfbvideosink: deal with rowstride as byte unit

rowstride is specified in bytes, but vio surface pitch is specified in
pixels. Add the necessary conversion to accomodate this.
---
 ext/directfb/dfbvideosink.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index d2fbe37..a20a5b0 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -2011,7 +2011,7 @@ gst_dfbvideosink_shvio_stretchblit (GstDfbVideoSink * dfbvideosink,
   } else {
 #endif /* defined(HAVE_SHMERAM) */
     if (dfbvideosink->rowstride >= 0)
-      viosurface[SRC].pitch = dfbvideosink->rowstride;
+      viosurface[SRC].pitch = byte2pixel (dfbvideosink->rowstride, src_format);
     else {
       viosurface[SRC].pitch = byte2pixel (src_pitch, src_format);
     }
-- 
1.7.10.4

